eis_toolkit/cli.py

```diff
@@ -236,15 +237,29 @@ def compute_pca_cli(
     """Compute principal components for the input data."""
     from eis_toolkit.exploratory_analyses.pca import compute_pca
 
-    typer.echo("Progress: 10%")
+    try:
+        geodataframe = gpd.read_file(input_vector)
+        dataframe = pd.DataFrame(geodataframe.drop(columns="geometry"))
 
-    geodataframe = gpd.read_file(input_vector)  # TODO: Check if gdf to df handling in tool itself
-    dataframe = pd.DataFrame(geodataframe.drop(columns="geometry"))
-    typer.echo("Progress: 25%")
+        output, variance_ratios = compute_pca(data=dataframe, number_of_components=number_of_components)
 
-    pca_df, variance_ratios = compute_pca(data=dataframe, number_of_components=number_of_components)
+    except:  # noqa: E722
+        try:
+            with rasterio.open(input_vector) as raster:
+                output, variance_ratios = compute_pca(data=raster, number_of_components=number_of_components)
+
+        except:  # noqa: E722
+            raise Exception("Could not read input file as raster or dataframe")
+
```

Comment:
Try with bare excepts is pretty bad and we should change this. I know this same pattern is used in another CLI func too, so we can go with this for now and refactor later (perhaps this file type recognition pattern refined deserves its own refined utility function – the other options are to separate vector and raster functions or parameterize whether we are dealing with a raster or vector).